synapse/crypto/keyclient.py

```diff
@@ -28,15 +30,15 @@
 
 logger = logging.getLogger(__name__)
 
-KEY_API_V1 = b"/_matrix/key/v1/"
+KEY_API_V2 = "/_matrix/key/v2/server/%s"
 
 
 @defer.inlineCallbacks
-def fetch_server_key(server_name, tls_client_options_factory, path=KEY_API_V1):
+def fetch_server_key(server_name, tls_client_options_factory, key_id):
     """Fetch the keys for a remote server."""
 
     factory = SynapseKeyClientFactory()
-    factory.path = path
+    factory.path = KEY_API_V2 % urllib.parse.quote(key_id)
```

Comment:
Generally I prefer to use an explicit tuple with %, in case your argument unexpectedly turns out to be a tuple. Agreed it's pretty unlikely here though.

```suggestion
    factory.path = KEY_API_V2 % (urllib.parse.quote(key_id), )
```

Reply:
That's true, and I prefer it too (I wish it was a syntax error, because sometimes I forget)